lib/icu.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+if (!process.binding('config').hasIntl)
+  throw new Error('This Node.js binary was built without ICU support.');
+
+const Buffer = require('buffer').Buffer;
+const normalizeEncoding = require('internal/util').normalizeEncoding;
+const icu = process.binding('icu');
+
+// Maps the supported transcoding conversions. The top key is the from_enc,
+// the child key is the to_enc. The value is the transcoding function to.
+const conversions = {
```

Comment:
It does not handle `hex` or `base64`. It should handle all the others. `internalUtil.normalizeEncoding()` is used under the covers.